src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/CaseActionAuditRepository.java

```diff
@@ -11,22 +11,69 @@
 @Repository
 public interface CaseActionAuditRepository extends JpaRepository<CaseActionAudit, Long> {
 
-    @Query("SELECT ca FROM case_action_audit ca "
-            + "WHERE (:userId IS NULL OR ca.userId = :userId) "
-            + "AND (:caseRef IS NULL OR ca.caseRef = :caseRef) "
-            + "AND (:caseTypeId IS NULL OR  ca.caseTypeId=:caseTypeId) "
-            + "AND (:caseJurisdictionId IS NULL OR ca.caseJurisdictionId=:caseJurisdictionId) "
-            + "AND (cast(:startTime as timestamp) IS NULL OR ca.timestamp >= :startTime) "
-            + "AND (cast(:endTime as timestamp) IS NULL OR ca.timestamp <= :endTime)")
+    @Query(value = "SELECT ca.* FROM case_action_audit ca "
+        + "WHERE (cast(:userId as text) IS NULL OR ca.user_id = cast(:userId as text)) "
+        + "AND (cast(:caseRef as text) IS NULL OR ca.case_ref = cast(:caseRef as text)) "
+        + "AND (cast(:caseTypeId as text) IS NULL OR ca.case_type_id = cast(:caseTypeId as text)) "
+        + "AND (cast(:caseJurisdictionId as text) IS NULL "
+        +   "OR ca.case_jurisdiction_id = cast(:caseJurisdictionId as text)) "
+        + "AND (cast(cast(:startTime as varchar) as timestamp) IS NULL "
+        +   "OR ca.log_timestamp >= cast(cast(:startTime as varchar) as timestamp)) "
+        + "AND (cast(cast(:endTime as varchar) as timestamp) IS NULL "
+        +   "OR ca.log_timestamp <= cast(cast(:endTime as varchar) as timestamp))",
+        countQuery = "SELECT count(*) FROM ( "
+            + "SELECT 1 FROM case_action_audit ca "
+            + "WHERE (cast(:userId as text) IS NULL OR ca.user_id = cast(:userId as text)) "
+            + "AND (cast(:caseRef as text) IS NULL OR ca.case_ref = cast(:caseRef as text)) "
+            + "AND (cast(:caseTypeId as text) IS NULL OR  ca.case_type_id = cast(:caseTypeId as text)) "
+            + "AND (cast(:caseJurisdictionId as text) IS NULL "
+            +   "OR ca.case_jurisdiction_id = cast(:caseJurisdictionId as text)) "
+            + "AND (cast(cast(:startTime as varchar) as timestamp) IS NULL "
+            +   "OR ca.log_timestamp >= cast(cast(:startTime as varchar) as timestamp)) "
+            + "AND (cast(cast(:endTime as varchar) as timestamp) IS NULL "
+            +   "OR ca.log_timestamp <= cast(cast(:endTime as varchar) as timestamp))"
+            + "limit 100000) ca",
+        nativeQuery = true)
     Page<CaseActionAudit> findCaseView(final @Param("userId") String userId,
                                        final @Param("caseRef") String caseRef,
                                        final @Param("caseTypeId") String caseTypeId,
                                        final @Param("caseJurisdictionId") String caseJurisdictionId,
                                        final @Param("startTime") Timestamp startTime,
                                        final @Param("endTime") Timestamp endTime,
                                        final Pageable pageable);
+
```

Comment:
I don’t think we need caseViewH2 version, I already have a single query in my PR that works for H2 too. I'll update this query tomorrow morning.